ksql-engine/src/main/java/io/confluent/ksql/planner/plan/DataSourceNode.java

```diff
@@ -100,24 +71,22 @@ public DataSourceNode(
     // DataSourceNode copies implicit and key fields into the value schema
     // It users a KS valueMapper to add the key fields
     // and a KS transformValues to add the implicit fields
-    this.schema = dataSource.getSchema()
-        .withMetaAndKeyFieldsInValue()
-        .withAlias(alias);
+    this.schema = StreamSource.getSchemaWithMetaAndKeyFields(alias, dataSource.getSchema());
```

Comment:
I 100% agree in principal. For this commit I'm just trying to preserve the fact that DataSourceNode.getSchema() returns an implementation schema because I have no idea what else in the analyzer/planner depends on that fact, and I'm not really confident in our tests to protect me there. I'm happy to tackle that problem in a follow-on - just trying to constrain scope as much as possible for this change. Otherwise I'll have a 5000 line PR :)

Reply:
Well, the QTT test have pretty good coverage now that _some_ tests include internal topic states as well.

However, you might want to look into enhancing QTT in a way I've been thinking about to improve coverage...

So at the moment we write a QTT test and define the inputs and expected outputs. However, the outputs don't have to include _all_ the topics.  So its possible between releases that the data stored on internal topics changes.  This would be a breaking change.

I've been thinking it might be nice to write a tool that can regenerate a QTT test with ALL the topic state captures in the expected output.  This would allow us to regenerate all our tests to cover all their state. 

This gives us much more confidence to change things as we'll know all state in Kafka is unchanged by the change, and its the state in Kafka that matters right?  (We could also do the same for schemas).

Such a tool is not throw away work. We've already discussed you splitting QTT into tests two tiers of tests. The first that check the physical model is built consistently for a given query and the second that tests a given physical model has a consistent output for a given input. Well, when doing this we'll want to add new tests to the second tier as things change - and those tests should capture and test _all_ output, so we can make use of this tool here.

I've not looked into how hard it would be to regen the tests with full output, but it might be worth the investment.